.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,211 @@
+      #     context: .
+      #     push: true
+      #     tags: ${{ env.IMAGE_NAME }}
+      - name: Build and save syncer image
+        run: |
+          set -x
+          docker build -t ${{ env.IMAGE_NAME }} .
+          docker save -o vcluster_syncer ${{ env.IMAGE_NAME }}
+      - name: Upload syncer image to artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: vcluster_syncer
+          path: ./vcluster_syncer
+          retention-days: 7
+
+  build-vcluster-cli:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Build vcluster cli
+        run: |
+          set -x
+          GO111MODULE=on GOFLAGS=-mod=vendor go build -v -o ./vcluster ./cmd/vclusterctl/main.go
+      - name: Upload vcluster cli to artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: vcluster
+          path: ./vcluster
+          retention-days: 7
+
+  get-testsuites-dir:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - id: set-paths-matrix
+        run: |
+          set -x
+          sudo apt-get install -y jq
+          paths=$(ls -d ./test/e2e*)
+          echo "::set-output name=matrix::$(printf '%s\n' "${paths}" | jq -R . | jq -cs .)"
+    outputs:
+      matrix: ${{ steps.set-paths-matrix.outputs.matrix }}
+
+  e2e-tests:
+    name: Execute test suites
+    needs: [build-and-push-syncer-image, build-vcluster-cli, get-testsuites-dir]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        distribution: ["k3s", "k8s", "k0s", "eks"]
+        test-suite-path: ${{fromJson(needs.get-testsuites-dir.outputs.matrix)}}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    
+    - name: Set up kind with K8s version v1.21.1
+      uses: engineerd/setup-kind@v0.5.0
+      with:
+        version: "v0.11.1"
+        image: kindest/node:v1.21.1
+    
+    - name: Testing kind cluster set-up
+      run: |
+          set -x
+          kubectl cluster-info
+          kubectl get pods -n kube-system
+          echo "kubectl config current-context:" $(kubectl config current-context)
+          echo "KUBECONFIG env var:" ${KUBECONFIG}
+    
+    - name: Download vcluster cli
+      uses: actions/download-artifact@v2
+      with:
+        name: vcluster
+    
+    - name: Download syncer image
+      uses: actions/download-artifact@v2
+      with:
+        name: vcluster_syncer
+
+    - name: Create vcluster
+      run: |
+        set -x 
+        
+        sudo apt-get install -y sed
+        
+        sed -i "s|REPLACE_IMAGE_NAME|${{ env.IMAGE_NAME }}|g" ${{ matrix.test-suite-path }}/../commonValues.yaml
+        
+        docker load < vcluster_syncer
+        
+        kind load docker-image ${{ env.IMAGE_NAME }}
```

Comment:
Nice to have: `docker load` could be skipped if we use `kind load image-archive`, but it would need to be tested to check that our `vcluster_syncer` file is in a compatible format.